src/sql/util.py

```diff
@@ -363,3 +383,39 @@ def show_deprecation_warning():
         "raise an exception in the next major release so please remove it.",
         FutureWarning,
     )
+
+
+def is_saved_snippet_or_table_exists(table, schema=None, task=None):
+    """
+    Check if the referenced table is a snippet
+    Parameters
+    ----------
+    table : str,
+        name of table
+
+    schema : str, default None
+        Name of the schema
+
+    task: str, default None
+        Command calling this function
+
+    Returns
+    -------
+    with_ : str, default None
+        name of the snippet
+    """
+    with_ = None
+    msg = "Analyzing"
+    if task == "explore":
+        msg = "Exploring"
+    elif task == "profile":
```

Comment:
also, I missed this in the last review. I think so many if's are looking a bit odd. and it's just adding `ing`. We can  put this in a dict {"explore":"exploring", ...} and fetch like dict[task].

Reply:
Is this resolved? @AnirudhVIyer

Reply:
Working on this!

Reply:
Have removed the messages part